core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeService.java

```diff
@@ -62,19 +67,20 @@ public DiscoveryNodeService addCustomAttributeProvider(CustomAttributesProvider
     }
 
     public DiscoveryNode buildLocalNode(TransportAddress publishAddress) {
-        Map<String, String> attributes = new HashMap<>(settings.getByPrefix("node.").getAsMap());
-        attributes.remove("name"); // name is extracted in other places
+        final String nodeId = generateNodeId(settings);
+        Map<String, String> attributes = new HashMap<>(Node.NODE_ATTRIBUTES.get(this.settings).getAsMap());
         if (attributes.containsKey("client")) {
-            if (attributes.get("client").equals("false")) {
-                attributes.remove("client"); // this is the default
-            } else {
-                // if we are client node, don't store data ...
-                attributes.put("data", "false");
-            }
+            throw new IllegalArgumentException("node.client setting is no longer supported, use " + Node.NODE_MASTER_SETTING.getKey()
```

Comment:
oh this is only possible because we allos `"node.*"` as a valid setting... grrr :)